extension/chrome/settings/modules/contacts.ts

```diff
@@ -23,6 +33,48 @@ Catch.try(async () => {
 
   BrowserMsg.listen(tabId); // set_css
 
+  const attUI = new AttUI(() => Promise.resolve({ size_mb: 5, size: 5 * 1024 * 1024, count: 1 }));
+  attUI.initAttDialog('fineuploader', 'fineuploader_button');
+  attUI.setAttAddedCb(async (file: Att) => {
+    (file !== null && typeof file !== 'undefined' && typeof file.id !== 'undefined' && file.id !== '') ? attUI.clearAtt(file.id) : attUI.clearAtts();
+    await handleKeyfileSelected(file);
+  });
+
+  const handleKeyfileSelected = async (file: Att) => {
+    try {
+      const rawFile = file.getData();
+      const stringFile = new TextDecoder("utf-8").decode(rawFile);
+      const { blocks } = await Pgp.armor.detectBlocks(stringFile);
+
+      const keys: OpenPGP.key.Key[] = [];
+      if (blocks !== null && blocks.length > 0 && blocks.some(block => block.type === "publicKey")) {
+        for (const block of blocks) {
+          if (block.type === "publicKey") {
+            const result = await openpgp.key.readArmored(block.content);
+            if (result.err !== null && typeof result.err !== 'undefined') {
+              throw result.err;
+            }
+            keys.push(...result.keys);
+          }
+        }
+      } else {
+        const result = await openpgp.key.read(rawFile);
+        if (result.err !== null && typeof result.err !== 'undefined') {
+          throw result.err;
```

Comment:
Again I would probably not catch errors here, just collect keys.

If you do catch errors, then you should be showing them to the user. `Ui.modal.error` can help you do that. As it is now, errors will just get reported to us (using handleErr) without informing the user, which is not useful anyway.

Reply:
So I changed it a bit. There is a little duplication now, which I can make a helper method for but it probably isn't required.

    if (result.err) {
        let e = `Got error processing armored key:\n`;
        result.err.map(v => e += `-> ${v.message}\n`);
        await Ui.modal.error(e);
        throw result.err;
    }

Reply:
You could collect errors the way you collect parsed keys into an erray, and then put this block once at the end.
Also I would not be re-throwing the error.